cmd/livepeer/livepeer.go

```diff
@@ -278,16 +278,18 @@ func main() {
 					glog.Fatalf("Unable to transcode using Nvidia gpu=%s err=%v", strings.Join(devices, ","), err)
 				}
 			}
-			// FIXME: Short-term hack to pre-load the detection model for the whole node
+			// FIXME: Short-term hack to pre-load the detection models on every device
 			if *sceneClassificationModelPath != "" {
 				detectorProfile := ffmpeg.DSceneAdultSoccer
 				detectorProfile.ModelPath = *sceneClassificationModelPath
 				core.DetectorProfile = &detectorProfile
-				tc, err := core.NewNvidiaTranscoderWithDetector(&detectorProfile, "0")
-				if err != nil {
-					glog.Fatalf("Could not initialize detector")
+				for _, d := range devices {
+					tc, err := core.NewNvidiaTranscoderWithDetector(&detectorProfile, d)
```

Comment:
Let's also update the transcoder initialization that happens at the start of every stream:
https://github.com/livepeer/go-livepeer/blob/fce8a401fbd3cc24dd6341a95921bd5e63b31001/core/transcoder.go#L149-L151
Currently it's hardcoded to device `0`, but it should now be able to use the same device `gpu` it is using for transcoding.